packages/types/src/getters/utils/create-getter.test.ts

```diff
@@ -0,0 +1,88 @@
+import { describe, expect, it } from 'vitest';
+import { createGetter } from './create-getter';
+
+interface Address {
+  city: string;
+  state: string;
+  country?: string;
+}
+
+interface Employee {
+  firstName: string;
+  lastName?: string;
+  address?: Address;
+}
+
+const employee: Employee = {
+  firstName: 'Alice',
+  address: {
+    city: 'San Francisco',
+    state: 'California',
+  },
+};
+
+const getFirstName = createGetter((employee?: Employee) => employee?.firstName);
+const getLastName = createGetter((employee?: Employee) => employee?.lastName);
+const getAddress = createGetter((employee?: Employee) => employee?.address);
+const getCity = createGetter((address?: Address) => address?.city);
+const getCountry = createGetter((address?: Address) => address?.country);
+const getFirstLetter = createGetter((value?: string) => value?.[0]);
```

Comment:
simple getters like this could be automatically created and attached to the return of a getter, eliminating some of the need for pipe syntax